### PR TITLE
BC-4143 - text by enter

### PR DESCRIPTION
**Pull request by christian-darsow**
# Short Description
Create a new text element as the first element and focus it when hitting enter key within title field

## Links to Ticket and related Pull-Requests
https://ticketsystem.dbildungscloud.de/browse/BC-4143

## Checklist before merging

- [x] QA: In addition to review, the code has been manually tested (if manual testing is possible)
- [x] PO: Any deviation from requirements was agreed with Product-Owner / ticket author / support-team

> Notice: Please keep this Pull-Request as a Draft (or add WIP label), until it is ready to be reviewed

### CODE REVIEW

**Review by odalys-dataport — DISMISSED** [review]

File: src/components/feature-board/shared/BoardAnyTitleInput.vue [hunk 1]
```diff
@@ -103,12 +111,23 @@ export default defineComponent({
 			}
 		});
 
+		const onEnter = ($event: KeyboardEvent) => {
+			if (props.scope !== "card") return;
+			if (!textarea.value) return;
+
+			if (textarea.value.value.length === textarea.value.selectionStart) {
```
Comment (MartinSchuhmacher) [inline src/components/feature-board/shared/BoardAnyTitleInput.vue:118]: `textarea.value.value` sounds really weird code wise. can we set `textarea.value` into another variable and make it more readable? also for line 116 then?

**Review by MartinSchuhmacher — CHANGES_REQUESTED** [review]

File: src/components/feature-board/shared/BoardAnyTitleInput.vue (hunk 1 above)
Comment (christian-darsow) [reply src/components/feature-board/shared/BoardAnyTitleInput.vue:118]: hm. the first `value` is for the reactive part. the second is then the property "value" of that reactive object ([HTMLTextAreaElement](https://developer.mozilla.org/en-US/docs/Web/API/HTMLTextAreaElement)). So somewhere in the code I guess we will always have that .value.value Do you have an idea how to make it more readable though?

Comment (MartinSchuhmacher) [reply src/components/feature-board/shared/BoardAnyTitleInput.vue:118]: if its like that then maybe something like `const reactiveTextarea = textarea.value` might fit as a first thought

Comment (christian-darsow) [reply src/components/feature-board/shared/BoardAnyTitleInput.vue:118]: To be honest, I don't like this idea very much. This `.value` syntax tells me that something is reactive. I always watch out for this syntax to know whether something is reactive or not - even before reading a vars name - what do you think about an alternative: a computed prop thats name states that it presents the textareas value length?

Comment (MartinSchuhmacher) [reply src/components/feature-board/shared/BoardAnyTitleInput.vue:118]: sounds good to me 👍

Comment (christian-darsow) [reply src/components/feature-board/shared/BoardAnyTitleInput.vue]: I moved the length calculation to a seperat function. This still needs that weird looking `.value.value` but makes it more readable by the function name I hope

Comment (MartinSchuhmacher) [reply src/components/feature-board/shared/BoardAnyTitleInput.vue]: I agree

**Pull request merged**